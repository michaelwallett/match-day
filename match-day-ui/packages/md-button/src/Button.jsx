import React, { Component } from 'react';
import styled from 'styled-components';
import colours from 'md-colours';

const StyledButton = styled.button`
  font-size: 1.5em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  background: ${(props) => props.primary ? colours.primary : colours.secondary};
  color: ${(props) => props.primary ? 'white' : colours.primary};
`;

export default class Button extends Component {
  render() {
    return <StyledButton {...this.props}>{this.props.children}</StyledButton>;
  }
}
