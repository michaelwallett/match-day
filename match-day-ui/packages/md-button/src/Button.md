Basic button:

    <Button>Push Me</Button>

Primary button:

    <Button primary>Push Me</Button>
