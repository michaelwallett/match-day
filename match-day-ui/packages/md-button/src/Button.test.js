import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('When basic button rendered', () => {
  test('should output button', () => {
    const component = renderer.create(
      <Button>Press Me</Button>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
