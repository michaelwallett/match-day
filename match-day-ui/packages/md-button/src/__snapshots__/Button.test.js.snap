// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When basic button rendered should output button 1`] = `
<button
  className="sc-bdVaJa fjIueO"
>
  Press Me
</button>
`;
